x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/slm/action/TransportExecuteSnapshotLifecycleAction.java

```diff
@@ -53,12 +53,7 @@ protected void masterOperation(final Task task, final ExecuteSnapshotLifecycleAc
                                    final ActionListener<ExecuteSnapshotLifecycleAction.Response> listener) {
         try {
             final String policyId = request.getLifecycleId();
-            SnapshotLifecycleMetadata snapMeta = state.metadata().custom(SnapshotLifecycleMetadata.TYPE);
```

Comment:
Why is this being removed?

Reply:
I collapsed it in into the second if -- notice that I added the `SnapshotLifecycleMetadata.EMPTY` as a second argument of `custom(...)`.

That is, prior to my change, the custom call could return `null`, so it was necessary to have this check twice -- if there's no `SnapshotLifecycleMetadata` then return an error, but then if there is a `SnapshotLifecycleMetadata` then check inside it to see if there's a `SnapshotLifecyclePolicyMetadata` and if there isn't one of those then also return the exact same error.

Now that I've made the call to `.custom(...)` null safe, just only the second if is sufficient.

Reply:
Ah, thanks for the explanation. I missed that. This cleanup in this file looks good now.